packages/sanity/src/studio/screens/schemaErrors/SchemaProblemGroups.tsx

```diff
@@ -0,0 +1,133 @@
+import {ErrorOutlineIcon, WarningOutlineIcon} from '@sanity/icons'
+import {Box, Breadcrumbs, Card, Flex, Stack, Text, ThemeColorToneKey} from '@sanity/ui'
+import {generateHelpUrl} from '@sanity/generate-help-url'
+import {SchemaValidationProblemGroup} from '@sanity/types'
+import React, {useMemo} from 'react'
+import styled from 'styled-components'
+
+const TONES: Record<'error' | 'warning', ThemeColorToneKey> = {
+  error: 'critical',
+  warning: 'caution',
+}
+
+const SegmentSpan = styled.code`
+  && {
+    background: none;
+    color: inherit;
+  }
+`
+
+const ErrorMessageText = styled(Text)`
+  white-space: pre-line;
+`
+
+export function SchemaProblemGroups(props: {problemGroups: SchemaValidationProblemGroup[]}) {
+  const {problemGroups} = props
+
+  const items = useMemo(() => {
+    const ret = []
+
+    for (const problemGroup of problemGroups) {
+      for (const problem of problemGroup.problems) {
+        ret.push({group: problemGroup, problem})
+      }
+    }
+
+    return ret
+  }, [problemGroups])
```

Comment:
Might be easier to use `React.memo` on the component here, since it is the only prop received?

Reply:
This was already there from earlier (this file has been slightly modified + renamed and git doesn't seem to have picked it up).